core/src/main/java/org/apache/brooklyn/entity/group/zoneaware/ProportionalZoneFailureDetector.java

```diff
@@ -56,4 +56,10 @@ protected boolean doHasFailed(Location loc, ZoneHistory zoneHistory) {
             return numDatapoints >= minDatapoints && proportionFailure >= maxProportionFailures;
         }
     }
+
+    @Override
+    public String toString(){
+        return getClass().getName();
```

Comment:
I'd go for the code below (using `com.google.common.base.Objects`):

    @Override
    public String toString() {
        return Objects.toStringHelper(this)
                .add("minDatapoints", minDatapoints)
                .add("timeToConsider",  timeToConsider)
                .add("maxProportionFailures", maxProportionFailures)
        .toString();
    }